### PR TITLE
ci: Replace prbot-stale with actions/stale

### LINKED ISSUE #18404: Replace stale bot with GH action
It looks that the stale bot hasn't been working for a while. See #12072 which should have been closed by now. We are not the only ones noticing this issue has there are some issues in the stale bot repository https://github.com/probot/stale/issues/354 already mentioning the same.

We should potentially replace it with this one https://github.com/marketplace/actions/close-stale-issues

cc @twpayne

### PR DESCRIPTION
**Pull request by twpayne**
Fixes #18404.

### CODE REVIEW

File: .github/workflows/close-stale-issues.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        - good-first-issue
+
+        days-before-pr-stale: 30
+        stale-pr-message: |
+          This pull request has been automatically marked as stale because it has not had
+          recent activity. It will be closed if no further activity occurs. Thank you
+          for your contributions.
+        days-before-pr-close: 14
+        close-pr-message: |
+          This issue has not seen any activity since it was marked stale. Closing.
```
Comment (aanm) [inline .github/workflows/close-stale-issues.yml]: `s/issue/pull request`

**Review by aanm — APPROVED** [review]
Just a small nit, otherwise looks good to me. Thank you!

File: .github/workflows/close-stale-issues.yml (hunk 1 above)
Comment (twpayne) [reply .github/workflows/close-stale-issues.yml]: Thanks, done.

**Review by nbusseneau — APPROVED** [review]
Thanks! LGTM. Had a run over parameters at https://github.com/actions/stale just in case, but I don't think we'd use anything else. We could potentially refactor parameters with the same values (e.g. `days-before-issue-close` and `days-before-pr-close` into `days-before-close`) but I think having them separated is better for future maintenance.

**Review by aanm — APPROVED** [review]

**Pull request merged**